rest-api/offline/table_exporter.py

```diff
@@ -0,0 +1,33 @@
+from google.appengine.api import app_identity
+from google.appengine.ext import deferred
+from offline.sql_exporter import SqlExporter
+from werkzeug.exceptions import BadRequest
+
+class TableExporter(object):
+  """API that exports data from our database to UTF-8 CSV files in GCS.
+
+  Used instead of Cloud SQL export because it handles newlines and null characters in a way that
+  other CSV clients (e.g. BigQuery, Google Sheets) can actually understand.
+  """
+
+  @classmethod
+  def _export_csv(cls, bucket_name, database, directory, table_name):
+    SqlExporter(bucket_name, use_unicode=True).run_export('%s/%s.csv' % (directory, table_name),
+                                                          'SELECT * FROM %s.%s' %
```

Comment:
This is a vector for injections -- obviously only from trusted clients, but this should be sanitized. Maybe an assert listing the allowed values for `database` and `table_name`?

Reply:
RDR and table_name are already validated below.

I added logic ensuring table_name matches a regex. I'd rather not hardcode the possible tables here, since that will be a maintenance issue over time.

Reply:
The regex is good enough -- but would rather bake it into this `_export_csv` function directly and apply to the database + table name, since it's so general. This ensures that nobody will accidentally call this function with unsanitized attributes in the future. 

The additional check on `database` below is also good :)